### PR TITLE
CSV Chores

### PR DESCRIPTION
**Pull request by rexagod**
Increase timeout, replace csv by subscription, and check CatalogSource
readiness.

### CODE REVIEW

File: test-prow-e2e.sh [hunk 1]
```diff
@@ -17,19 +17,23 @@ echo "Enabling Console Plugin for ODF Operator"
...
+export ODF_CSV_NAME="$(oc get subscription -n openshift-storage -o=jsonpath='{.items[?(@.metadata.name=="odf-operator")].status.installedCSV}')"
 
 oc patch csv ${ODF_CSV_NAME} -n openshift-storage --type='json' -p \
   "[{'op': 'replace', 'path': '/spec/install/spec/deployments/1/spec/template/spec/containers/0/image', 'value': \"${ODF_CONSOLE_IMAGE}\"}]"
 
 # Installation occurs.
 
 # This is also the default case if the CSV is in "Installing" state initially.
-timeout 500 bash <<-'EOF'
+timeout 15m bash <<-'EOF'
```
Comment (bipuladh) [inline test-prow-e2e.sh:28]: Poll rather than arbitrary waits.

Comment (vbnrh) [reply test-prow-e2e.sh:28]: @bipuladh Like pranshu mentioned . It is an upper limit to avoid endless waits

File: test-prow-e2e.sh [hunk 2]
```diff
@@ -17,19 +17,23 @@ echo "Enabling Console Plugin for ODF Operator"
...
 echo "waiting for ${ODF_CSV_NAME} clusterserviceversion to succeed"
 until [ "$(oc -n openshift-storage get csv -o=jsonpath="{.items[?(@.metadata.name==\"${ODF_CSV_NAME}\")].status.phase}")" == "Succeeded" ]; do
   sleep 1
 done
+echo "waiting for odf-catalogsource connection to be in READY state"
+until [ "$(oc get catalogsource -n openshift-marketplace -o=jsonpath='{.items[?(@.metadata.name=="odf-catalogsource")].status.connectionState.lastObservedState}')" == "READY" ]; do
+  sleep 1
+done
```
Comment (vbnrh) [inline test-prow-e2e.sh:36]: Move this to the top. This needs to be checked at the start

File: test-prow-e2e.sh [hunk 3]
```diff
@@ -17,19 +17,23 @@ echo "Enabling Console Plugin for ODF Operator"
 oc patch console.v1.operator.openshift.io ${CONSOLE_CONFIG_NAME} --type=json -p="[{'op': 'add', 'path': '/spec/plugins', 'value':["${ODF_PLUGIN_NAME}"]}]"
 
 ODF_CONSOLE_IMAGE="$1"
-export ODF_CSV_NAME="$(oc get csv -n openshift-storage -o=jsonpath='{.items[?(@.spec.displayName=="OpenShift Data Foundation")].metadata.name}')"
+export ODF_CSV_NAME="$(oc get subscription -n openshift-storage -o=jsonpath='{.items[?(@.metadata.name=="odf-operator")].status.installedCSV}')"
```
Comment (vbnrh) [inline test-prow-e2e.sh:20]: This needs another wait. It won't be populated initially

**Review by vbnrh — CHANGES_REQUESTED** [review]

File: test-prow-e2e.sh (hunk 3 above)
Comment (rexagod) [reply test-prow-e2e.sh:30]: Added wait here: https://github.com/red-hat-storage/odf-console/pull/59/files#diff-bf3590087e59ba96ab84c865a37047c28148e7cc0139d06de34ca53a282b0642R24

File: test-prow-e2e.sh (hunk 2 above)
Comment (rexagod) [reply test-prow-e2e.sh]: Moved here: https://github.com/red-hat-storage/odf-console/pull/59/files#diff-bf3590087e59ba96ab84c865a37047c28148e7cc0139d06de34ca53a282b0642R21

File: test-prow-e2e.sh (hunk 1 above)
Comment (bipuladh) [reply test-prow-e2e.sh]: okay!

File: test-prow-e2e.sh [hunk 4]
```diff
@@ -16,46 +16,70 @@ export ODF_PLUGIN_NAME="odf-console"
...
-INSTALLER_DIR=${INSTALLER_DIR:=${ARTIFACT_DIR}/installer}
 
 function generateLogsAndCopyArtifacts {
-  oc get storageclusters ocs-storagecluster -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagecluster.yaml
-  oc get storagesystems -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagesystem_details.yaml
-  oc get nodes -o yaml > ${ARTIFACT_DIR}/node.yaml
-  oc get csvs -n openshift-storage > ${ARTIFACT_DIR}/csvs.yaml
+  oc cluster-info dump > ${ARTIFACT_DIR}/cluster_info.json
+  oc get catalogsource -n openshift-storage -o wide > ${ARTIFACT_DIR}/catalogsource.yaml
+  oc get catalogsource -n openshift-storage -o yaml > ${ARTIFACT_DIR}/catalogsource.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:43]:
```suggestion
  oc get catalogsource -n openshift-storage -o yaml > ${ARTIFACT_DIR}/catalogsource.yaml
```
```suggestion
  oc get catalogsource -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/catalogsource.yaml
```
for appending

File: test-prow-e2e.sh [hunk 5]
```diff
@@ -16,46 +16,70 @@ export ODF_PLUGIN_NAME="odf-console"
...
+  oc get csvs -n openshift-storage -o wide > ${ARTIFACT_DIR}/csvs.yaml
   oc get csvs -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/csvs.yaml
-  oc get pods -n openshift-storage -o wide > ${ARTIFACT_DIR}/pods_details.yaml
-  oc get pods -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/pod_details.yaml
   oc get deployments -n openshift-storage -o wide > ${ARTIFACT_DIR}/deployment_details.yaml
   oc get deployments -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/deployment_details.yaml
+  oc get installplan -n openshift-storage -o wide > ${ARTIFACT_DIR}/installplan.yaml
+  oc get installplan -n openshift-storage -o yaml > ${ARTIFACT_DIR}/installplan.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:49]:
```suggestion
  oc get installplan -n openshift-storage -o yaml > ${ARTIFACT_DIR}/installplan.yaml
```
```suggestion
  oc get installplan -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/installplan.yaml
```

File: test-prow-e2e.sh [hunk 6]
```diff
@@ -16,46 +16,70 @@ export ODF_PLUGIN_NAME="odf-console"
...
+  oc get nodes -o wide > ${ARTIFACT_DIR}/node.yaml
+  oc get nodes -o yaml > ${ARTIFACT_DIR}/node.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:51]:
```suggestion
  oc get nodes -o yaml > ${ARTIFACT_DIR}/node.yaml
```
```suggestion
  oc get nodes -o yaml >> ${ARTIFACT_DIR}/node.yaml
```

File: test-prow-e2e.sh [hunk 7]
```diff
@@ -16,46 +16,70 @@ export ODF_PLUGIN_NAME="odf-console"
...
+  oc get pods -n openshift-storage -o wide > ${ARTIFACT_DIR}/pods_details.yaml
+  oc get pods -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/pod_details.yaml
+  oc get storageclusters ocs-storagecluster -n openshift-storage -o wide > ${ARTIFACT_DIR}/storagecluster.yaml
+  oc get storageclusters ocs-storagecluster -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagecluster.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:55]:
```suggestion
  oc get storageclusters ocs-storagecluster -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagecluster.yaml
```
```suggestion
  oc get storageclusters ocs-storagecluster -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/storagecluster.yaml
```

File: test-prow-e2e.sh [hunk 8]
```diff
@@ -16,46 +16,70 @@ export ODF_PLUGIN_NAME="odf-console"
...
+  oc get storagesystems -n openshift-storage -o wide > ${ARTIFACT_DIR}/storagesystem_details.yaml
+  oc get storagesystems -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagesystem_details.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:57]:
```suggestion
  oc get storagesystems -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagesystem_details.yaml
```
```suggestion
  oc get storagesystems -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/storagesystem_details.yaml
```

File: test-prow-e2e.sh [hunk 9]
```diff
@@ -16,46 +16,70 @@ export ODF_PLUGIN_NAME="odf-console"
...
+  oc get subscriptions -n openshift-storage -o wide > ${ARTIFACT_DIR}/subscription_details.yaml
   oc get subscriptions -n openshift-storage -o yaml > ${ARTIFACT_DIR}/subscription_details.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:59]:
```suggestion
  oc get subscriptions -n openshift-storage -o yaml > ${ARTIFACT_DIR}/subscription_details.yaml
```
```suggestion
  oc get subscriptions -n openshift-storage -o yaml > >${ARTIFACT_DIR}/subscription_details.yaml
```

**Review by vbnrh — CHANGES_REQUESTED** [review]

File: test-prow-e2e.sh [hunk 10]
```diff
@@ -17,45 +27,61 @@ echo "Enabling Console Plugin for ODF Operator"
...
 SCREENSHOTS_DIR=gui-test-screenshots
-INSTALLER_DIR=${INSTALLER_DIR:=${ARTIFACT_DIR}/installer}
 
 function generateLogsAndCopyArtifacts {
-  oc get storageclusters ocs-storagecluster -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagecluster.yaml
-  oc get storagesystems -n openshift-storage -o yaml > ${ARTIFACT_DIR}/storagesystem_details.yaml
-  oc get nodes -o yaml > ${ARTIFACT_DIR}/node.yaml
-  oc get csvs -n openshift-storage > ${ARTIFACT_DIR}/csvs.yaml
+  oc cluster-info dump > ${ARTIFACT_DIR}/cluster_info.json
+  oc get catalogsource -n openshift-storage -o wide > ${ARTIFACT_DIR}/catalogsource.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:42]:
```suggestion
  oc get catalogsources -A  > ${ARTIFACT_DIR}/catalogsource.yaml
```
can check all resources for this

File: test-prow-e2e.sh [hunk 11]
```diff
@@ -17,45 +27,61 @@ echo "Enabling Console Plugin for ODF Operator"
...
-  oc get subscriptions -n openshift-storage -o yaml > ${ARTIFACT_DIR}/subscription_details.yaml
-  oc cluster-info dump > ${ARTIFACT_DIR}/cluster_info.json
+  oc get installplan -n openshift-storage -o wide > ${ARTIFACT_DIR}/installplan.yaml
+  oc get installplan -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/installplan.yaml
+  oc get nodes -o wide > ${ARTIFACT_DIR}/node.yaml
+  oc get nodes -o yaml >> ${ARTIFACT_DIR}/node.yaml
+  oc get pods -n olm -o wide > ${ARTIFACT_DIR}/pods_details.yaml
+  oc get pods -n olm -o yaml >> ${ARTIFACT_DIR}/pod_details.yaml
+  oc get pods -n openshift-storage -o wide >> ${ARTIFACT_DIR}/pods_details.yaml
+  oc get pods -n openshift-storage -o yaml >> ${ARTIFACT_DIR}/pod_details.yaml
```
Comment (vbnrh) [inline test-prow-e2e.sh:55]: can segregate olm and openshift pods into different files

**Review by vbnrh — CHANGES_REQUESTED** [review]

**Pull request merged**